File: tests/test_files/sample.py
# This is a sample Python file for testing the parser.
import math
from collections import defaultdict

class MyClass:
    """A sample class."""
    def __init__(self, name):
        self.name = name

    def greet(self):
        """A sample method."""
        print(f"Hello, {self.name}")
        top_level_function(1, 2)

def top_level_function(arg1, arg2):
    """A top-level function."""
    return math.sqrt(arg1) + arg2

File: tests/test_files/sample.go
// Sample Go file for parser testing.
package main

import "fmt"

// Widget is a sample struct.
type Widget struct {
	Name  string
	Value int
}

func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

func add(a, b int) int {
	return a + b
}
